src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -1248,9 +1248,40 @@ public QueryOperationResponse<T> LoadProperty<T>(object entity, string propertyN
         public Uri GetReadStreamUri(object entity)
         {
             Util.CheckArgumentNull(entity, "entity");
-            EntityDescriptor box = this.entityTracker.GetEntityDescriptor(entity);
+            EntityDescriptor box = GetEntityDescriptorForStreamEntity(entity);
             return box.ReadStreamUri;
         }
+        /// <summary>
+        /// This function is used to get the entity descriptor that is to be checked when trying to resolve streams.
+        /// For other use cases use the <see cref="GetEntityDescriptor"/> function
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        private EntityDescriptor GetEntityDescriptorForStreamEntity(object entity)
+        {
+            EntityDescriptor box;
+            if (MergeOption == MergeOption.NoTracking)
+            {
+                BaseEntityType baseEntity = entity as BaseEntityType;
+                if (baseEntity == null)
+                {
+                    throw Error.InvalidOperation(Strings.Context_EntityMediaLinksNotTrackedInEntity);
+                }
+
+                box = baseEntity?.EntityDescriptor;

```

Comment:
```suggestion
                box = baseEntity.EntityDescriptor;
```
The null check has already happened in line 1266